packages/functional-tests/tests/syncV3/signIn.spec.ts

```diff
@@ -20,129 +22,186 @@ test.describe('severity-2 #smoke', () => {
-      await signinTokenCode.successMessage.waitFor({ state: 'visible' });
       await expect(signinTokenCode.successMessage).toBeVisible();
       await expect(signinTokenCode.successMessage).toContainText(
         /Email re-?sent/
       );
       const code = await target.email.waitForEmail(
         email,
-        EmailType.verifyShortCode,
-        EmailHeader.shortCode
+        EmailType.verifyLoginCode,
+        EmailHeader.signinCode
       );
       await signinTokenCode.input.fill(code);
-      await login.submit();
+      await signinTokenCode.submit.click();
+
       await expect(connectAnotherDevice.fxaConnected).toBeVisible();
     });
 
-    test('verified - invalid code', async ({
+    test('verified email with signin verification, accepts valid sign in code', async ({
       emails,
       target,
       syncBrowserPages,
     }) => {
-      const { configPage, page, login, connectAnotherDevice, signinTokenCode } =
+      const { page, login, connectAnotherDevice, signinTokenCode } =
         syncBrowserPages;
+
       const [email] = emails;
-      const config = await configPage.getConfig();
-      test.fixme(
-        config.showReactApp.signUpRoutes,
-        'this test goes through the signup flow instead of sign in, skipping for react, see FXA-8973'
+
+      // Simulate a new sign up that requires a signin verification code.
+      target.auth.signUp(email, PASSWORD, {
+        service: 'sync',
+        preVerified: 'true',
+      });
+
+      await page.goto(
+        `${target.contentServerUrl}?context=fx_desktop_v3&service=sync&action=email`
       );
+      await login.login(email, PASSWORD);
+
+      const code = await target.email.waitForEmail(
+        email,
+        EmailType.verifyLoginCode,
+        EmailHeader.signinCode
+      );
+      await signinTokenCode.input.fill(code);
+      await signinTokenCode.submit.click();
+
+      await expect(connectAnotherDevice.fxaConnected).toBeVisible();
+    });
+
+    test('verified email with signin verification, rejects invalid signin code', async ({
+      emails,
+      target,
+      syncBrowserPages,
+    }) => {
+      const { page, login, connectAnotherDevice, signinTokenCode } =
+        syncBrowserPages;
+
+      const [email] = emails;
+
+      // Simulate a new sign up that requires a signin verification code.
+      target.auth.signUp(email, PASSWORD, {
+        service: 'sync',
+        preVerified: 'true',
+      });
+
       await page.goto(
         `${target.contentServerUrl}?context=fx_desktop_v3&service=sync&action=email`
       );
-      await login.setEmail(email);
-      await signinTokenCode.clickSubmitButton();
-      await login.setPassword(PASSWORD);
-      await login.confirmPassword(PASSWORD);
-      await login.setAge('21');
-      await login.submit();
+      await login.login(email, PASSWORD);
 
       // Input invalid code and verify the tooltip error
       await signinTokenCode.input.fill('000000');
```

Comment:
The old test likely landed on the `confirm_signup_code` page - new path check in the POM should ensure we're on the `signin_token_code` page as expected now 🎉